### ISSUE TITLE
command: Add Timeout resource parameter

### ISSUE THREAD

**Issue by clintoncwolfe** [inspec/inspec#3772]
### 🎛 Description

🙋 feature request

This issue is a narrow-scope case of #1675.

Add the ability for a `command` resource to time out.

If it does time out, currently fail the resource, but this should be an Unable resource case per #3174.

Proposed syntax:

```
# Timeout in seconds
describe command('something_awful --really-slow', timeout: 30) do
  #...
end
```

If the command finishes in less than 30 seconds, the resource should finish in less than 30 seconds (in other words, don't wait out the watchdog).

If the command takes less than 1 second, it is acceptable to finish in a bit more than 1 second (in other words, the watchdog may check in one-second intervals, not subsecond).

### 🌍 InSpec and Platform Version

3.4.x

Requested backport to 1.x

Feature is requested for RHEL 5,6,7; then Windows (versions TBD), then AIX.

### 💁 Possible Solutions

Ruby's Timeout is considered [dangerous](https://medium.com/@adamhooper/in-ruby-dont-use-timeout-77d9d4e5a001).

We might be able to use an OS facility for timing out a process.

We might be able to wrap the invocation in [a shell script](https://stackoverflow.com/questions/687948/timeout-a-command-in-bash-without-unnecessary-delay).

We might look into the underlying Ruby library we use to shell out (might just be train) and see if there is already a timeout feature for commands there.

What was done on #2288?

Aha! Link: https://chef.aha.io/features/SH-2294

**Comment by james-stocks** [comment]
See also https://github.com/inspec/train/issues/352 for timing out in train.
Would need to be careful not to leave terminated connections hanging; and also not to leave a bad entry in the backend cache, etc

**Comment by clintoncwolfe** [comment]
Fixed in #5443
